### PR TITLE
test(v1.7.0): Phase 3 Part C — T-2 / T-3 / T-8 + macOS CI (T-12) — closes umbrella #32

### PR DESCRIPTION
**Pull request by sgwannabe**
## Summary

ComBba Phase 3 Part C — 5 final Test & CI Coverage items from umbrella #32. Closes umbrella with 1 deferral (T-7 → post-hackathon, see ASSESSMENT.md).

### T-2 — \`_filled_ratio\` reference + CI fixture
- \`scripts/compute-filled-ratio.py\`: single source of truth for the 9-slot rule in \`idea-spec.schema.json\`. Implements binary nested-object slot rule, 'unknown' filtering, array-length ≥1 rule.
- \`tests/fixtures/filled-ratio/cases.json\` — **7 cases** covering: B-3 skip (1/9 ≈ 0.1111), B-1 fast path (5/9 ≈ 0.5556), full path (9/9 = 1.0), nested-all-null, leaf-'unknown', empty-array, bool-false.
- Verify script asserts script output matches expected to 4 decimal places.

### T-3 — \`must_have_constraints\` normalize + schema enum
- \`scripts/normalize-constraints.py\`: maps user-facing labels per idea-clarifier.md Batch C table. Free-form input → \`type: \"other\"\`.
- Schema \`must_have_constraints.items.type\` gains \`enum: [\"regulatory\",\"budget\",\"latency\",\"team_size\",\"platform\",\"data_residency\",\"other\"]\` (7 canonical).
- \`tests/fixtures/normalize-constraints/cases.json\` — **13 cases** including canonical buckets with/without inline value, freeform Korean/English, edge cases (empty, whitespace, mismatched parens, uppercase).
- All existing fixtures (rule9-fp-guard, runs/_fixture) continue to validate.

### T-8 — LESSON 0.7 regression fixture
- New \`tests/fixtures/lesson07-regression/\` proves Rule 9 anchors on \`chosen_preview\` (user's H1 pick), not panel composite_winner.
- **2 cases**: respect P10 user pick when composite was P02 (exit 0); block drift toward P02 Slack-bot vocabulary when chosen_preview is P10 (exit 2).
- Closes coverage gap from LESSON 0.7's narrative regression scenario.

### T-12 — macOS CI matrix (Windows deferred)
- \`verify-plugin\` job converted to \`[ubuntu-latest, macos-14]\` matrix. macOS catches BSD-vs-GNU userland divergence (mktemp/sed/find) that ubuntu-only CI missed in PR #45.
- Windows defer rationale in ASSESSMENT.md.

### T-7 — deferred (assessment in \`tests/fixtures/ASSESSMENT.md\`)
- Mock UX harness for AskUserQuestion replay: 1–2 day build cost + sharp maintenance risk during agent-prompt iteration. Marginal value over the 5 artifact fixtures is moderate, not critical for hackathon.
- Re-open trigger documented (Socratic-flow regression slipping past artifact fixtures).

## CI integration
\`\`\`bash
bash tests/fixtures/security/verify-security.sh           # 13 sections
bash tests/fixtures/rule9-fp-guard/verify-rule9.sh        # 3 cases
bash tests/fixtures/filled-ratio/verify-filled-ratio.sh   # 7 cases  ← NEW
bash tests/fixtures/normalize-constraints/verify-normalize.sh  # 13 cases  ← NEW
bash tests/fixtures/lesson07-regression/verify-lesson07.sh # 2 cases  ← NEW
\`\`\`

All 5 suites green on local Ubuntu macOS pre-push.

## Closes
- Umbrella #32 P3 Test & CI Coverage (T-1/T-4/T-5/T-6/T-9/T-10/T-11 already shipped; T-2/T-3/T-8/T-12 in this PR; T-7 deferred per ASSESSMENT.md).

## Remaining work
- P8 Requirements Expansion (umbrella #37, 9 items) — last v1.7 audit umbrella.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **새로운 기능**
  * macOS 환경에서 CI 테스트 실행 지원 추가
  * 제약 조건 정규화 기능 추가

* **테스트**
  * 채움 비율 검증 테스트 추가
  * 제약 조건 정규화 테스트 추가
  * Lesson 0.7 회귀 테스트 추가

* **문서**
  * Phase 3 평가 및 진행 상황 문서 추가

### CODE REVIEW

File: tests/fixtures/filled-ratio/verify-filled-ratio.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  exit 1
+fi
+
+echo "=== T-2 _filled_ratio verify ==="
+echo
+
+python3 -c "import json; json.load(open('$FIXTURES_DIR/cases.json'))" >/dev/null \
+  || { echo "x cases.json malformed" >&2; exit 1; }
+
+case_count=$(python3 -c "import json; print(len(json.load(open('$FIXTURES_DIR/cases.json'))))")
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply tests/fixtures/filled-ratio/verify-filled-ratio.sh]: Applied in 477ef54 — $FIXTURES_DIR를 python3 -c 문자열에 보간하지 않고 sys.argv[1]로 전달하도록 정정. 모든 3개 신규 fixture 스크립트(verify-filled-ratio/verify-normalize/verify-lesson07) 동일 패치. 3개 모두 local green 재확인. PR scope 외 기존 fixture는 별도 issue로 추적 권장.

File: tests/fixtures/lesson07-regression/verify-lesson07.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+[[ -x "$HOOK" || -r "$HOOK" ]] || { echo "x idea-drift-detector.py missing at $HOOK" >&2; exit 1; }
+
+echo "=== T-8 LESSON 0.7 regression verify ==="
+echo
+
+python3 -c "import json; json.load(open('$FIXTURES_DIR/cases.json'))" >/dev/null \
+  || { echo "x cases.json malformed" >&2; exit 1; }
+
+case_count=$(python3 -c "import json; print(len(json.load(open('$FIXTURES_DIR/cases.json'))))")
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply tests/fixtures/lesson07-regression/verify-lesson07.sh]: Applied in 477ef54 — 동일 패치. 3개 fixture 스크립트 모두 sys.argv[1] 패턴으로 통일. shell injection surface 제거.

File: tests/fixtures/normalize-constraints/verify-normalize.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+[[ -r "$SCRIPT" ]] || { echo "x normalize-constraints.py missing at $SCRIPT" >&2; exit 1; }
+
+echo "=== T-3 must_have_constraints normalize verify ==="
+echo
+
+python3 -c "import json; json.load(open('$FIXTURES_DIR/cases.json'))" >/dev/null \
+  || { echo "x cases.json malformed" >&2; exit 1; }
+
+case_count=$(python3 -c "import json; print(len(json.load(open('$FIXTURES_DIR/cases.json'))))")
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply tests/fixtures/normalize-constraints/verify-normalize.sh]: Applied in 477ef54 — 동일 패치. tr -dc allowlist sanitize는 추가하지 않음 (argv 전달이 진짜 방어이고 sanitize는 paranoid theater). 신규 3개 fixture 모두 동일 변경.

File: tests/fixtures/lesson07-regression/verify-lesson07.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+# For each case in cases.json, materialise the run workspace
+# (chosen_preview.json + idea.spec.json) and pipe an Edit/Write hook
+# input describing the incoming content. Assert the observed exit code
+# matches expected_exit.
+#
+# Goal: prove that Rule 9 anchors on chosen_preview (user's H1 pick),
+# not on the panel composite_winner — the regression LESSON 0.7
+# warns about.
+
+set -u
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply tests/fixtures/lesson07-regression/verify-lesson07.sh]: Applied in b115847 — verify-lesson07.sh를 `set -euo pipefail`로 변경. set +e/-e wrapping이 의도대로 (hook 종료코드만 잡고 그 외는 errexit) 동작. 신규 3개 fixture 스크립트 전부 동일하게 strict mode 통일 (verify-filled-ratio, verify-normalize). 3개 모두 local green 재확인.
